### PR TITLE
Add LimitRanger plugin to labels, taints and annotations

### LINKED ISSUE #31994: Annotation kubernetes.io/limit-ranger not documented
**This is a Bug Report**

**Problem:** Kubernetes uses an annotation, `kubernetes.io/limit-ranger`; however, this annotation is not registered or documented.

**Proposed Solution:**
Document the annotation

**Page to Update:**
https://kubernetes.io/docs/reference/labels-annotations-taints/

**Additional Information:**
/language en

### PR DESCRIPTION
**Pull request by PriyanshuAhlawat**
closes #31994

### CODE REVIEW

**Comment by Sea-n** [comment]
Seems reasonable
/lgtm

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 1]
```diff
@@ -114,6 +114,19 @@ to the name of the namespace. You can't change this label's value.
 This is useful if you want to target a specific namespace with a label
 {{< glossary_tooltip text="selector" term_id="selector" >}}.
 
+### kubernetes.io/limit-ranger
+
+Example: `kubernetes.io/limit-ranger: LimitRanger plugin set: cpu, memory request for container nginx; cpu, memory limit for container nginx`
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:119]: For clarity:
```suggestion
Example: `kubernetes.io/limit-ranger: "LimitRanger plugin set: cpu, memory request for container nginx; cpu, memory limit for container nginx"`
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 2]
```diff
@@ -114,6 +114,19 @@ to the name of the namespace. You can't change this label's value.
...
+
+Used on: Namespace
+
+Kubernetes by default doesn't provide any resource limit, that means unless you explicitly define limits,
+your container can consume unlimited CPU and memory.
+We can define default limit for pods by creating a LimitRange in the relevant namespace.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:125]:
```suggestion
You can define a default request or default limit for pods. You do this by creating a LimitRange
in the relevant namespace.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 3]
```diff
@@ -114,6 +114,19 @@ to the name of the namespace. You can't change this label's value.
...
+Pods deployed after this LimitRange will have these limits applied to them.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:126]:
```suggestion
Pods deployed after you define a LimitRange will have these limits applied to them.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 4]
```diff
@@ -114,6 +114,19 @@ to the name of the namespace. You can't change this label's value.
...
+The annotation `limit-ranger` indicates that limits were requested for the pod and they were applied successfully.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:127]:
```
The annotation `kubernetes.io/limit-ranger` indicates that limits were specified for the Pod,
and they were applied successfully.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 5]
```diff
@@ -114,6 +114,19 @@ to the name of the namespace. You can't change this label's value.
...
+For more details - [LimitRange](/docs/concepts/policy/limit-range)
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:128]:
```suggestion
For more details, read about [LimitRanges](/docs/concepts/policy/limit-range).
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 6]
```diff
@@ -114,6 +114,19 @@ to the name of the namespace. You can't change this label's value.
 This is useful if you want to target a specific namespace with a label
 {{< glossary_tooltip text="selector" term_id="selector" >}}.
 
+### kubernetes.io/limit-ranger
+
+Example: `kubernetes.io/limit-ranger: LimitRanger plugin set: cpu, memory request for container nginx; cpu, memory limit for container nginx`
+
+Used on: Namespace
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:121]:
```suggestion
Used on: Pod
```

**Review by sftim — COMMENTED** [review]
I can see several things that seem incorrect here.

/lgtm cancel

**Comment by natalisucks** [comment]
@PriyanshuAhlawat Hi there – there have been several suggestions left after a review, do you mind looking into these so we could move forward with this PR? Thanks!

**Comment by PriyanshuAhlawat** [comment]
@natalisucks I have applied the suggested changes, now you can take a look 🙂

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 7]
```diff
@@ -114,6 +114,20 @@ to the name of the namespace. You can't change this label's value.
...
+
+Used on: Pod
+
+Kubernetes by default doesn't provide any resource limit, that means unless you explicitly define limits,
+your container can consume unlimited CPU and memory.
+You can define a default request or default limit for pods. You do this by creating a LimitRange in the relevant namespace.
+Pods deployed after you define a LimitRange will have these limits applied to them.
+The annotation `kubernetes.io/limit-ranger` records that resource defaults were specified for the Pod,
+and they were applied successfully.
+For more details, read about [LimitRanges](/docs/concepts/policy/limit-range).
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:129]: Ah. I suspect that this annotation might also be applied on a PersistentVolumeClaim. If so, we ought to add that detail here before we merge it.

@SergeyKanzhelev do you have an opinion here?

Comment (SergeyKanzhelev) [reply content/en/docs/reference/labels-annotations-taints/_index.md:129]: yes, I noticed that PVC was added to the document as a quick fix. I'd prefer we address it as a separate PR.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Review by sftim — APPROVED** [review]
Thanks

We prefer _not_ to make whitespaces changes to other text in PRs (_separate_ cleanup PRs are very very welcome).

/approve

**Comment by sftim** [comment]
PriyanshuAhlawat, thank you for working on this.

**Pull request merged**